app/uk/gov/hmrc/merchandiseinbaggage/controllers/EoriNumberController.scala

```diff
@@ -35,12 +36,13 @@ class EoriNumberController @Inject()(
   actionProvider: DeclarationJourneyActionProvider,
   override val repo: DeclarationJourneyRepository,
   view: EoriNumberView,
-  mibConnector: MibConnector
+  mibConnector: MibConnector,
+  navigator: Navigator
 )(implicit ec: ExecutionContext, appConfig: AppConfig)
     extends DeclarationJourneyUpdateController with Logging {
 
   private def backButtonUrl(implicit request: DeclarationJourneyRequest[_]) =
-    backToCheckYourAnswersIfCompleteElse(routes.CustomsAgentController.onPageLoad())
+    backToCheckYourAnswersIfCompleteElse(CustomsAgentController.onPageLoad())
```

Comment:
potential to extend the navigator to have .previousPage() as well

Reply:
sure, i was going to do as part of second iteration with backlink too